Sources/MongoSwift/BSON/Document.swift

```diff
@@ -20,8 +20,15 @@ public class DocumentStorage {
     }
 }
 
-/// A struct representing the BSON document type.
+/// A struct representing the BSON document type. `Document`s are append-only and do not support
+/// overwriting or removing keys in-place, or provide safeguards against duplicate keys. 
+/// To transform documents, use `Sequence` protocol methods along with `filter` and `mapValues`.
+/// To check for existing keys, use the `.keys` property.
 public struct Document: ExpressibleByDictionaryLiteral, ExpressibleByArrayLiteral {
+
+    /// The element type of a document: a tuple containing an individual key-value pair.
+    public typealias Element = (key: String, value: BsonValue?)
```

Comment:
Is an "Element" type alias a Swift convention?

Reply:
so the `Sequence` protocol in Swift has an `associatedtype Element` requirement ([docs](https://developer.apple.com/documentation/swift/sequence)). 
The compiler was actually inferring this already based on the type we return from `DocumentIterator.next()`, but inspired by the source code for Swift `Dictionary` etc., I think explicitly declaring it is nice for clarity, and also for giving us a shorthand way to refer to this specific tuple type.